### DISCUSSION TITLE
Composter improvements

### DISCUSSION THREAD

**Discussion by happyburger23** [Roadhog360/Et-Futurum-Requiem#280] [Improvements]
- Hoppers should be able to insert plants into the composter
- Particles that appear when items are inserted should be at the top of the composter as it is for vanilla MC

EDIT: Comparators should also read the composter's fullness level, also vanilla MC

<img width="842" alt="Screenshot 2023-05-22 at 7 46 33 PM" src="https://github.com/Roadhog360/Et-Futurum-Requiem/assets/93412223/4a43a918-75a2-440f-93f4-a1ff179824be">

**Comment by Swifty07** [comment]
is the hopper thing from vanilla

**Comment by GregoriusT** [comment]
Yeah in vanilla Hoppers can interact with Composters, to the point that Hoppers literally lag LESS if they have a Composter above them, but I am pretty sure you need to mixin/asm something into the Hopper itself to make it work unless you add a TileEntity to the Composter.

**Reply by happyburger23** [reply]
Comparators can also read the composter's fullness level

**Comment by Roadhog360** [comment]
This was fulfilled in [ce20886](https://github.com/Roadhog360/Et-Futurum-Requiem/commit/ce20886ccdc68bb841e6f68b3f9eecaa3c1057b6).
